src/mds/journal.cc

```diff
@@ -1620,8 +1620,10 @@ void EMetaBlob::replay(MDSRank *mds, LogSegment *logseg, int type, MDPeerUpdate
 	if (used_preallocated_ino) {
 	  if (!session->info.prealloc_inos.empty()) {
 	    inodeno_t ino = session->take_ino(used_preallocated_ino);
-	    session->info.prealloc_inos.erase(ino);
 	    ceph_assert(ino == used_preallocated_ino);
+	    if (session->info.prealloc_inos.contains(ino)) {
+	      session->info.prealloc_inos.erase(ino);
+	    }
```

Comment:
assertion before erasing the inode from prealloc_inos is intentional, this will help figure out that the `used_preallocated_ino` is not present in `prealloc_inos`

Reply:
This will just make the assert more graceful, but still couldn't avoid the MDS crashing. Except this we still need to figure out why it couldn't find the `used_preallocated_ino` in the `session->info.prealloc_inos` to avoid crashing if possible.

Reply:
this is just sort of a better handling, not a solution

Reply:
@lxbsz you mean if the `used_preallocated_ino` is not present in the `prealloc_inos` then we re-spawn?

Reply:
what happens in case the `used_preallocated_ino` is corrupted? if the corruption is handled somewhere, its fine else we are just going to be put in a endless loop

Reply:
Yeah, it's possible. This is why I said `if it's really journal logs are partially flushed` issue.

While as we discussed before this commit only improves the code and will make the crash more graceful instead of resolving this issue.